src/App.tsx

```diff
@@ -1,93 +1,243 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import classNames from 'classnames';
+import {
+  useEffect,
+  useRef,
+  useState,
+  useContext,
+} from 'react';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { PatchedTodo, Todo } from './types/Todo';
+import {
+  deleteTodo,
+  getTodos,
+  postTodo,
+  updateTodo,
+  changeTodos,
+} from './api/todos';
+import { Errors } from './types/Errors';
+import { FilterType } from './types/FilterType';
+import {
+  LoadingContext,
+} from './components/LoadingContext/LoadingContext';
+import { LoadingSpinner } from './components/Loader';
+
+const USER_ID = 10209;
+
+const getFilteredTodos = (todos: Todo[], filterType: string) => {
+  switch (filterType) {
+    case FilterType.ALL:
+      return todos;
+
+    case FilterType.ACTIVE:
+      return todos.filter((todo) => !todo.completed);
+
+    case FilterType.COMPLETED:
+      return todos.filter((todo) => todo.completed);
+
+    default:
+      return [];
+  }
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<Errors>(Errors.NONE);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.ALL);
+  const [title, setTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [uploadingTodos, setUploadingTodos] = useState(false);
+
+  const { setIsLoadingAll } = useContext(LoadingContext);
+
+  const isLoading = useRef(false);
```

Comment:
Wow, why `isLoading` it's ref? Also, I didn't find any places where it's needed, can we remove it?